### ISSUE TITLE
Implement emitRound and emitCompareOp for LLVMGenerator.ArithmeticLLVMGenerator

### ISSUE THREAD

**Issue by johanvos** [oracle/graal#1168]
In PR #1157, LLVMGenerator.ArithmeticLLVMGenerator now implementends AArch64ArithmeticGenerator. 
The methods `emitRound` and `emitCompareOp` are not yet implemented.

This could be done differently though, by moving the `emitCountLeadingZeros` and `emitCountTralingZeros` to ArithmeticLLVMGenerator so that the dependency on an AArch64 specific interface is not needed in ArithmeticLLVMGenerator.

**Comment by vjovanov** [comment]
@loicottet has this been fixed?

**Comment by loicottet** [comment]
No, the LLVMArithmeticGenerator is still dependent on the AArch64-specific stuff. A long-term fix will come from the refactoring of lowerings so that code using the LLVM backend only has nodes corresponding to LLVM intrinsics, instead of the platform-specific ones like today.

**Comment by hamzaGhaissi** [comment]
Looks like `emitCountLeadingZeros` and `emitCountTralingZeros` have been moved to `ArithmeticLLVMGenerator`
https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core.graal.llvm/src/com/oracle/svm/core/graal/llvm/LLVMGenerator.java#L1540
https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core.graal.llvm/src/com/oracle/svm/core/graal/llvm/LLVMGenerator.java#L1547

**Comment by loicottet** [comment]
`emitRound` was also moved to `ArithmeticLIRGeneratorTool`, and implementing it could be an interesting optimization. It is however not a priority at the moment, since `RoundNode` is not emitted on the LLVM backend and rounding is correctly handled by other means.
On the other hand, `emitCompareOp` has indeed remained a target-specific operation and `ArithmeticLLVMGenerator` is now target-independent, so it is not an issue anymore.
